projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -1491,7 +1495,6 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
      * @deprecated you should use onGridKeydown event
      */
     @Output()
-    @DeprecateProperty('onFocusChange event is deprecated. Use onGridKeydown event instead.')
```

Comment:
I believe the `@DeprecateProperty` decorator is still needed. While the warning does not show up on compile, it does not show up when it should.
With these changes, for example, if a user creates a grid with:
```html
<igx-grid (onFocusChange)="myChangeHandler($event)">
...
</igx-grid>
```
the deprecation warning **will not** show. It will show only when the emit is called.

I think that the issue can be resolved a bit cleaner. Continuing the example with `onFocusChange`. The current implementation
```typescript
export class IgxGridBaseComponent {
// Declaration
@DeprecateProperty(...)
public onFocusChange = new EventEmitter<IFocusChangeEventArgs>();
...
// Call accesses deprecated property
this.onFocusChange.emit(...);
}
```
**should become**
```typescript
export class IgxGridBaseComponent {
// Declaration
private _onFocusChange = new EventEmitter<IFocusChangeEventArgs>();
...
// Public accessor w/ deprecation warning
@DeprecateProperty(...)
public get onFocusChange(): EventEmitter<IFocusChangeEventArgs> {
   return this._onFocusChange;
}
public set onFocusChange(val: EventEmitter<IFocusChangeEventArgs>) {
   this._onFocusChange = val;
} 
...
// Call **does not** access deprecated property
this._onFocusChange.emit(...);
}
```
This way, the internal calls will bypass the `@DeprecateProperty` decorator (by calling the `private` property) while the `public` API of the component will remain unchanged (attempting to access the deprecated properties will still throw a warning).

Eventually, when the deprecated property is removed, all of these are just deleted.

Please consider this and let me know if it makes sense.